src/components/popover/popover.test.jsx

```diff
@@ -0,0 +1,106 @@
+import React, { useRef, Fragment } from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import Popover from './popover.jsx';
+
+describe('Popover', () => {
+  function PopoverWithToggle(popoverProps) {
+    const popoverRef = useRef();
+
+    return (
+      <Fragment>
+        <button onClick={() => { popoverRef.current.open = true; }}>Open Popover</button>
+        <button onClick={() => { popoverRef.current.open = false; }}>Close Popover</button>
+        <span>Some text</span>
+        <Popover ref={popoverRef} {...popoverProps}>{popoverProps.children}</Popover>
+      </Fragment>
+    );
+  }
+
+  it('has defaults', () => {
+    const popoverRef = React.createRef();
+
+    render(<Popover title="Defaults" />);
+    expect(document.body).toMatchSnapshot();
+
+    render(<Popover title="Defaults" startOpen={true} />);
+    expect(document.body).toMatchSnapshot();
+
+    render(<Popover ref={popoverRef} title="Defaults with ref" />);
+    expect(popoverRef.current).toMatchSnapshot();
+  });
+
+  it('has a title', () => {
+    render(<PopoverWithToggle title="Another title" />);
+
+    userEvent.click(screen.getByText('Open Popover'));
+    expect(screen.getByText('Another title').tagName).toEqual('H1');
```

Comment:
It's probably more user-centric to make an assertion on the label.

Something like: 

```suggestion
    expect(screen.getByRole('dialog', { name: 'Another title' })).toBeInTheDocument();
```